### ISSUE TITLE
iOS-Profile -"Hmm, it's not here" pg when background/reopen app if select existing secondary contact

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32938]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.11-13
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Log in with gmail account
3. Tap settings >> Profile >> Contact method
4. Enter secondary contact which is already a existing user (Note: Red dot is displayed)
5. Select the secondary contact (Note: Error message displayed)
6. Background & foreground app

## Expected Result:
After background & reopen it user still see the error message page.
No "Hmm, it's not here" page displayed

## Actual Result:
"Hmm, it's not here" page when background & reopen app if select existing secondary contact

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/a78c0a9d-4d37-4043-aeed-ed3a7761073b

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- iOS-Profile -"Hmm, it's not here" pg when background/reopen app if select existing secondary contact
### What is the root cause of that problem?
- After refreshing the page, app will call OpenApp API which will reset the loginList. Then, all the error contact method is removed.
- That leads to ```!contactMethod || !loginData``` in the bellow is true, so the FullPageNotFoundPage is displayed:
https://github.com/Expensify/App/blob/207e9ded0504e760e0b298944a1369efb72e3bc4/src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js#L235
### What changes do you think we should make in order to solve the problem?
- We need the fix from BE side that does not remove all the error contact methods in OpenApp api.
- And we should navigate user to the previous screen once the ```!contactMethod || !loginData``` is true

### What alternative solutions did you explore? (Optional)
- NA

**Comment by DylanDylann** [comment]
@Ollyws Beside the BE change, should we navigate user to the previous screen if  ```!contactMethod || !loginData```: true```?

**Comment by abekkala** [comment]
@lanitochka17 on this page once you're prompted with the red error message are you clicking Delete? (as we have noted that it's not possible to add that address as a secondary)
<img width="285" alt="Screenshot 2023-12-18 at 10 12 33 AM" src="https://github.com/Expensify/App/assets/11966460/6788c225-c56a-4243-a74f-7882c8a655cd">

What exactly are you clicking/doing to get the Hmm, its not here page?

**Comment by lanitochka17** [comment]
@abekkala Hello
I don’t press anything, after the background  open the app again

**Comment by DylanDylann** [comment]
@abekkala The issue also can be reproduced in web platform

**Comment by abekkala** [comment]
@robertjchen can you confirm that this would require a back end work first? https://github.com/Expensify/App/issues/32938#issuecomment-1858172481

Also, not sure if this aligns with any wave work right now.

**Comment by abekkala** [comment]
@robertjchen just a friendly bump here - before continuing on this one can we confirm that it'll require some internal back end work first? https://github.com/Expensify/App/issues/32938#issuecomment-1858172481

**Comment by abekkala** [comment]
@twisterdotcom 
### STATUS:

We're at the beginning and no proposals have been submitted/accepted. 

Waiting on confirmation from internal engineer (@robertjchen) that some back end work would need to be completed before continuing with this job.

**Comment by robertjchen** [comment]
> We need the fix from BE side that we should not remove all the error contact methods in OpenApp api.

I'm not sure if changing the backend will help here. Unless this information is sent to the backend, the `loginList` that is returned in OpenApp would never contain the information entered- is my understanding correct? 🤔

**Comment by DylanDylann** [comment]
@robertjchen As i mentioned in my proposal:

> After refreshing the page (or Background & foreground mobile app), it will call OpenApp API which will reset the loginList. Then, all the error contact methods are removed.

**Comment by robertjchen** [comment]
> it will call OpenApp API which will reset the loginList

Could you please be more specific on what you would like the backend to do? Does this means that `loginList` will need to be passed to the backend? Or do we need `OpenApp` to not return an empty `loginList`?

**Comment by DylanDylann** [comment]
@robertjchen 
> After refreshing the page (or Background & foreground mobile app), it will call OpenApp API which will reset the loginList. Then, all the error contact methods are removed.

I want that the OpenApp API returns the error contact methods as well

**Comment by robertjchen** [comment]
> I want that the OpenApp API returns the error contact methods as well
I'm not sure how this is possible without the App sending this information to the backed in the first place 🤔 

So, I think the proper answer here is that the OpenApp API call should NOT replace the existing `loginList` in Onyx.

**Comment by DylanDylann** [comment]
- @Ollyws @twisterdotcom  Based on comment [here](https://github.com/Expensify/App/issues/32938#issuecomment-1872376672), we cannot make the BE change. So should we navigate user to the previous screen as I mentioned in the [proposal](https://github.com/Expensify/App/issues/32938#issuecomment-1852598334)?

**Comment by twisterdotcom** [comment]
It looks like @robertjchen has started a PR to resolve this internally, so I think we should just make this internal actually, right @robertjchen?

**Comment by robertjchen** [comment]
Correct, re-labeling for now

**Comment by robertjchen** [comment]
Actually on second thought, I think @DylanDylann 's proposal [here](https://github.com/Expensify/App/issues/32938#issuecomment-1852598334) should work fine given this specific flow. 🤔  It's simple and should address what is a bit of a corner case without additional complexity in both the frontend and backend.

@Ollyws Let me know if that sounds good, if so- let's move forward with this.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by DylanDylann** [comment]
- @robertjchen @Ollyws  Actually, this bug can still be reproduced based on the RCA [here](https://github.com/Expensify/App/issues/32938#issuecomment-1852598334) 

- We have a lot of cases that lead to the contact method with error. And in this case, in the latest main, we do not allow user to save the duplicate contact method anymore, so we can not follow the reproduce steps to reproduce the bug, based on [comment](https://github.com/Expensify/App/issues/32938#issuecomment-1886340989). 

- But, we still can reproduce by another action that leads to the error contact method

**Comment by DylanDylann** [comment]
@robertjchen Please help me, how to create the error contact method? (what data should I send to BE)

**Comment by robertjchen** [comment]
I think the idea was that we would create an entirely new API call that sends whatever the user entered as the error contact, since it'll be returned back to the app after it's resumed (via the response of `OpenApp`).

However, I think it's a lot of work just to save that field- can't we just return to the previous page? The user will just re-enter the error contact again after it's resumed, we just want to avoid the "Hmm, it's not here" page

**Comment by DylanDylann** [comment]
@robertjchen 

> can't we just return to the previous page

Yes. This is my plan. But, I mean, currently, this bug cannot be reproduced, because we prevent users from inputting the existing contact method from FE side

**Comment by abekkala** [comment]
@robertjchen do you have any further feedback for @DylanDylann irt their proposal?

**Comment by abekkala** [comment]
@robertjchen @Ollyws is the proposal above sufficient to proceed or should I pull in someone to work on this?

**Comment by robertjchen** [comment]
Yeah, given the specific flow, I'm not sure what else we can help with from the backend. Let's close out for now and revisit if needed 👍
